libid/fractals.cpp

```diff
@@ -728,3 +733,29 @@ int other_julia_fp_per_pixel()
     }
     return 0;
 }
+
+int burning_ship_fp_fractal()
+{
+    DComplex z, q;
+    double real_imag;
+    int degree = (int) g_params[2];
+    q.x = g_init.x;
+    q.y = g_init.y;
+    if (degree == 2) // Burning Ship
+    {
+        g_temp_sqr_x = sqr(g_old_z.x);
```

Comment:
All math functions from the standard library should be qualified as being in namespace `std` and `<cmath>` included to obtain their declarations.